concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState.hs

```diff
@@ -897,10 +897,11 @@ doAddressWouldClash pbs addr = do
         bsp <- loadPBS pbs
         Accounts.addressWouldClash addr (bspAccounts bsp)
 
-doRegIdExists :: (IsProtocolVersion pv, MonadBlobStore m) => PersistentBlockState pv -> ID.CredentialRegistrationID -> m (Maybe AccountIndex)
+doRegIdExists :: (IsProtocolVersion pv, MonadBlobStore m) => PersistentBlockState pv -> ID.CredentialRegistrationID -> m Bool
+
 doRegIdExists pbs regid = do
         bsp <- loadPBS pbs
-        fst <$> Accounts.regIdExists regid (bspAccounts bsp)
+        isJust . fst <$> Accounts.regIdExists regid (bspAccounts bsp)
```

Comment:
Yes that is quite possible. But as the complexity of `lookup` and `member` are the same, I didn't see any reason to change that with this PR.
I only did the `Bool` thing as that was what i needed in the `TransactionVerifier`. So I didn't see any harm in leaving it be as it could be possible we needed that `AccountIndex` later on.

I can change it though.